dist/bin/dotr

```diff
@@ -31,18 +31,49 @@ source "$PROG_HOME/bin/common"
 
 CLASS_PATH=".$PSEP$DOTTY_LIB$PSEP$SCALA_LIB"
 
-# -d must be the first option
-case "$1" in
-  -d) DEBUG="$DEBUG_STR" && shift ;;
-esac
 
-first_arg="$1"
+addResidual () {
+  residual_args+=("$1")
+}
 
-if [ -z "$1" ]; then
+addCustomClassPath () {
+  classpath_args+=("$1:")
+}
+
+run_repl=false
+
+while [[ $# -gt 0 ]]; do
+key=$1
+case $key in
+        -repl)
+         run_repl=true
+         shift
+         ;;
+        -classpath)
+          addCustomClassPath "$2"
+          shift
+          shift
+          ;;
+        -d)
+        DEBUG="$DEBUG_STR"
+         shift
+         ;;
+         *)
+           addResidual "$1"
+           shift
+           ;;
+
+  esac
+done
+
+CLASS_PATH="${classpath_args[@]}$CLASS_PATH"
+
+if [ $run_repl == true ]; then
+    echo "Starting dotty REPL"
+    echo "$PROG_HOME/bin/dotc -classpath $CLASS_PATH -repl ${residual_args[@]}"
```

Comment:
Here the classpath should only contain `classpath_args` if it is not empty.

What does it mean exactly `classpath_args[@]`? I'm not familiar with the `[@]` part.

Reply:
@nicolasstucki Afaik, the `[*]` or `[@]` subscripts differ only within double quotes, with `[*]` each element inside the array is seperated by `IFS`